File: mart/src/main/java/shopping/mart/dto/ProductResponse.java
package shopping.mart.dto;

public final class ProductResponse {

    private Long id;
    private String name;
    private String imageUrl;
    private String price;

    public ProductResponse() {
    }

    public ProductResponse(final Long id, final String name, final String imageUrl, final String price) {
        this.id = id;
        this.name = name;
        this.imageUrl = imageUrl;
        this.price = price;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public String getPrice() {
        return price;
    }

    @Override
    public String toString() {
        return "ProductResponse{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", imageUrl='" + imageUrl + '\'' +
                ", price='" + price + '\'' +
                '}';
    }
}

File: src/main/kotlin/shopping/controller/model/ProductResponse.kt
package shopping.controller.model

import shopping.repository.model.Product

data class ProductResponse(
    val id: Long,
    val name: String,
    val price: Int,
    val imageUrl: String,
) {
    companion object {
        fun from(product: Product): ProductResponse =
            with(product) {
                ProductResponse(
                    id = id!!,
                    name = name,
                    price = price,
                    imageUrl = imageUrl,
                )
            }
    }
}
